lib/precompiler.js

```diff
@@ -4,7 +4,17 @@ import fs from 'fs';
 import * as Handlebars from './handlebars';
 import {basename} from 'path';
 import {SourceMapConsumer, SourceNode} from 'source-map';
-import uglify from 'uglify-js';
+
+// We are using `require` instead of `import` here, because es6-modules do not allow
+// dynamic imports and uglify-js is an optional dependency.
+var uglify;
+try {
+  uglify = require('uglify-js');
+} catch (err) {
+  if (err.code !== 'MODULE_NOT_FOUND') {
+    throw err;
+  }
+}
```

Comment:
@nknapp it might be easier to move the `require()` call into the `if (opts.min) {` condition block below. that way you will only try to import the dependency if someone is actually trying to use the `min` option and you won't need the try-catch at all.